sparrow-py/pysrc/sparrow_py/sources/arrow.py

```diff
@@ -0,0 +1,59 @@
+"""Provide sources based on PyArrow, including Pandas and CSV."""
+from io import StringIO
+from typing import Optional
+from typing import Union
+
+import pandas as pd
+import pyarrow as pa
+
+from .source import Source
+
+
+class ArrowSource(Source):
+    """A source reading data batches from Arrow or Pandas."""
+
+    def __init__(
+        self,
+        time_column_name: str,
+        key_column_name: str,
+        data: Union[pd.DataFrame, pa.RecordBatch],
```

Comment:
Probably a good idea to subclass this eventually, for people who aren't familiar with Arrow and are looking for how to get their Pandas data in.

Reply:
Makes sense. I was debating that while writing this PR.

Looking around, I actually wonder if this should be `_arrow.py` since people should import from `sparrow_py.sources`.